sway/config/output.c

```diff
@@ -125,6 +126,63 @@ void merge_output_config(struct output_config *dst, struct output_config *src) {
 	}
 }
 
+struct output_config *duplicate_output_config(const struct output_config *src) {
+	struct output_config *oc = calloc(1, sizeof(struct output_config));
+	if (oc == NULL) {
+		return NULL;
+	}
+	oc->name = strdup(src->name);
+	if (oc->name == NULL) {
+		free(oc);
+		return NULL;
+	}
+	oc->enabled = src->enabled;
+	oc->width = src->width;
+	oc->height = src->height;
+	oc->refresh_rate = src->refresh_rate;
+	oc->custom_mode = src->custom_mode;
+	oc->x = src->x;
+	oc->y = src->y;
+	oc->scale = src->scale;
+	oc->scale_filter = src->scale_filter;
+	oc->transform = src->transform;
+	oc->subpixel = src->subpixel;
+	oc->max_render_time = src->max_render_time;
+	oc->adaptive_sync = src->adaptive_sync;
+	if (src->background) {
+		oc->background = strdup(src->background);
+	}
+	if (src->background_option) {
+		oc->background_option = strdup(src->background_option);
+	}
+	if (src->background_fallback) {
+		oc->background_fallback = strdup(src->background_fallback);
+	}
+	oc->dpms_state = src->dpms_state;
+	return oc;
+}
+
+bool compare_output_config(const struct output_config *dst, const struct output_config *src) {
```

Comment:
I'm not super thrilled about having yet another list of output config fields to maintain.

An alternative approach would be to have some kind of `config_dirty` field, set to true when changing a field. But finding all spots where we update a field doesn't sound fun either.